testutils/sampleoptions.go

```diff
@@ -0,0 +1,95 @@
+package testutils
+
+import (
+	"errors"
+
+	"github.com/kyma-project/nats-manager/api/v1alpha1"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+type SampleOption func(*unstructured.Unstructured) error
```

Comment:
In which context the `SampleOption` is used? Also the name does not seem familiar to me, could you please explain the reason behind that name?

Reply:
This is only used in tests, to provide options in test helpers like `NewSampleNATSStatefulSetUnStruct`. There is no reason behind that name. If you have any suggestion for alternate name, let me know?

Reply:
I don't get the name `Sample`, maybe it can be more clear when we advance more in development.